kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/SaslInspectionIT.java

```diff
@@ -0,0 +1,310 @@
+import org.assertj.core.api.InstanceOfAssertFactory;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import io.github.nettyplus.leakdetector.junit.NettyLeakDetectorExtension;
+
+import io.kroxylicious.proxy.config.ConfigurationBuilder;
+import io.kroxylicious.proxy.config.NamedFilterDefinition;
+import io.kroxylicious.proxy.config.NamedFilterDefinitionBuilder;
+import io.kroxylicious.proxy.testplugins.ClientAuthAwareLawyerFilter;
+import io.kroxylicious.proxy.testplugins.ClientTlsAwareLawyer;
+import io.kroxylicious.proxy.testplugins.ProtocolCounter;
+import io.kroxylicious.proxy.testplugins.ProtocolCounterFilter;
+import io.kroxylicious.proxy.testplugins.SaslInspection;
+import io.kroxylicious.test.assertj.KafkaAssertions;
+import io.kroxylicious.testing.kafka.api.KafkaCluster;
+import io.kroxylicious.testing.kafka.common.BrokerConfig;
+import io.kroxylicious.testing.kafka.common.SaslMechanism;
+import io.kroxylicious.testing.kafka.junit5ext.KafkaClusterExtension;
+import io.kroxylicious.testing.kafka.junit5ext.Topic;
+
+import static io.kroxylicious.test.tester.KroxyliciousConfigUtils.proxy;
+import static io.kroxylicious.test.tester.KroxyliciousTesters.kroxyliciousTester;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(KafkaClusterExtension.class)
+@ExtendWith(NettyLeakDetectorExtension.class)
+class SaslInspectionIT {
+    // client handshakes with PLAIN
+    // proxy and broker have PLAIN enabled
+    // client authenticated with the correct password
+    // => client should be able to produce and consume
+    @Test
+    // @formatter:off
+    void shouldAuthenticateWhenSameMechanism_PLAIN_withReauth(
+                                                              @SaslMechanism(value = "PLAIN", principals = {
+                                                                      @SaslMechanism.Principal(user = "alice", password = "alice-secret")
+                                                              })
+                                                              @BrokerConfig(name = "connections.max.reauth.ms", value = "5000")
+                                                              KafkaCluster cluster,
+                                                              Topic topic)
+            throws Exception {
+        // @formatter:on
+
+        String mechanism = "PLAIN";
+        String clientLoginModule = "org.apache.kafka.common.security.plain.PlainLoginModule";
+        String username = "alice";
+        String password = "alice-secret";
+
+        assertClientsCanAccessCluster(cluster, topic, mechanism, clientLoginModule, username, password,
+                2, 1,
+                10_000);
+    }
+
+    // client handshakes with PLAIN
+    // proxy and broker have PLAIN enabled
+    // client authenticated with the correct password
+    // => client should be able to produce and consume
+    @Test
+    void shouldAuthenticateWhenSameMechanism_PLAIN(
```

Comment:
I think you could reduce the duplication in this test by using your funky @ConstraintsMethodSource to permutate the SASL arguments.